lua/litee/symboltree/init.lua

```diff
@@ -338,6 +338,21 @@ function M.dump_node()
     lib_tree.dump_tree(ctx.node)
 end
 
+local select_config_key = function(key, user_config, default_config)
+  local val = user_config[key]
+  local default_val = config[key]
+
+  if nil == val then
```

Comment:
Should we do this? For "keymaps" values, user cannot provide "nil" then.

Reply:
Would you like to move it to the place "only for tables" i.e.
```diff
   local default_val = config[key]
 
-  if nil == val then
-    return default_val
-  end
-
   if "table" ~= type(default_val) then
     return val
   end
 
+  if nil == val then
+    return default_val
+  end
+
   return vim.tbl_extend("keep", val, default_val)
 end
```
?

Reply:
The logic of this loop should be:

"Did user provide key?"
"If yes, replace default_config[key] with user_config[key]" (allow nil)
"If no, make no modifications to default_config"

Reply:
Ah, I see, now we have a nested table, Should we write a function which handles the parsing of "keymap" table itself? We'd need to put in the default values if user does not provide all keys in their keymap field.

Reply:
To fully make keymap parsing working, what if we do the following?

-> If user_config["keymap"] == {} ; map no buffer local key maps
-> If user_config["keymap"] == nil; map default key maps
-> if user_config["keymap"] == { some_values_populated }; set default_config["keymap"] = user_config["keymap"]

does this sound OK?

Reply:
> -> If user_config["keymap"] == {} ; map no buffer local key maps 
In such case the window could be driven from the command line only and we would have to wrap `nvim_buf_set_keymap` calls with `if` to disable keymaps (if I understood what you mean). 

With https://github.com/ldelossa/litee-symboltree.nvim/pull/11#discussion_r783260757 `nil` has no special meaning (as without this PR).

Reply:
How do we support "do not map any buffer local keymaps"

Reply:
Nohow. In `main` local (buffer) mappings are defined every time so disabling them would require additional `if`. 

> -> If user_config["keymap"] == {} ; map no buffer local key maps

I suspect that my English does not allow me to fully understand something about your comments.

Reply:
So I think we *should* somehow allow users to map *no* buffer local mappings if they do not want to. Do you know of a way to accomplish this? 

My first idea is to check number of table keys in `user_config["keywords"]` - if number of keys == 0 (user provided an empty map {}) we use this to map *no* buffer key mappings. 

Its okay to wrap `nvim_buf_set_keymap` functions in an `if` statement to do this.

Reply:
It sounds similarly to the choice "does the user need to have the set of global commands or not" mentioned in #7 ;)

> Do you know of a way to accomplish this?

Probably
```lua
 --  if "table" == type(default_val)
  if nil == val then
    return nil
  end
```
and in `buffer` 
```lua
if nil ~= config.keymaps then
    vim.api.nvim_buf_set_keymap(buf, "n", config.keymaps.expand, ":LTExpandSymboltree<CR>", opts)
 -- ...
```

Reply:
LGTM, in #7 when you said "commands" i thought you meant user commands like "LTOpenSymbolTree"